### ISSUE TITLE
[Bug]: Server crash immediately following FTB Backups job

### ISSUE THREAD

**Issue by ukmcplyr** [FTBTeam/FTB-Modpack-Issues#1935]
### Modpack

FTB Stoneblock 3

### Modpack version

1.2.0

### Log Files

[latest.log](https://pste.ch/gujeluwefo.sql)
[debug.log](https://pste.ch/ocoyegenov.md)
[crash log #1](https://pste.ch/avofejupis.php)
[crash log #2](https://pste.ch/enociyerob.pl)

### Describe the bug

One of the servers I'm playing on crashed whilst seemingly in the middle of an FTB Backups job.

```
[14Nov2022 00:00:00.166] [ftbbackups2_Worker-1/ERROR] [net.minecraft.server.level.ChunkMap/]: Failed to save chunk 3,0
net.minecraft.ReportedException: Accessing PalettedContainer from multiple threads
	at net.minecraft.util.ThreadingDetector.m_199417_(ThreadingDetector.java:88) ~[server-1.18.2-20220404.173914-srg.jar%23252!/:?]
	at net.minecraft.util.ThreadingDetector.m_199422_(ThreadingDetector.java:69) ~[server-1.18.2-20220404.173914-srg.jar%23252!/:?]
	at net.minecraft.world.level.chunk.PalettedContainer.m_63120_(PalettedContainer.java:42) ~[server-1.18.2-20220404.173914-srg.jar%23252!/:?]
[...]
```

The crash happened when I was clearing an area with a Mining Gadget with the Light Placer module enabled. Possibly related to [MC-162080](https://bugs.mojang.com/browse/MC-162080).

Re-connecting and everything seems fine, though the client now complains incorrectly that the trashslot mod is missing from the server:

```
[14Nov2022 00:44:09.687] [Render thread/INFO] [net.blay09.mods.trashslot.TrashSlot/]: TrashSlot is not installed on the server and thus will be unavailable.
```

```
$> l mods/trashslot-forge-1.18.1-11.0.2.jar; grep trashslot logs/*.log |head -n3
204264769 -rw-r--r-- 1 minecraft minecraft 110K Feb 22  2022 mods/trashslot-forge-1.18.1-11.0.2.jar
logs/debug.log:[14Nov2022 00:06:34.673] [main/DEBUG] [net.minecraftforge.fml.loading.moddiscovery.ModFileParser/LOADING]: Considering mod file candidate /home/minecraft/servers/ftb/stoneblock3/1.2.0/mods/trashslot-forge-1.18.1-11.0.2.jar
logs/debug.log:[14Nov2022 00:06:34.674] [main/DEBUG] [net.minecraftforge.fml.loading.moddiscovery.ModFileInfo/LOADING]: Found valid mod file trashslot-forge-1.18.1-11.0.2.jar with {trashslot} mods - versions {11.0.2}
logs/debug.log:[14Nov2022 00:06:34.882] [main/DEBUG] [net.minecraftforge.fml.loading.moddiscovery.AbstractJarFileDependencyLocator/]: Failed to load resource META-INF/jarjar/metadata.json from trashslot-forge-1.18.1-11.0.2.jar, it does not contain dependency information
```
### Steps to reproduce

Unsure as yet if reproducable, will report back if I can get it to crash again.

### Expected behaviour

No crash

**Comment by impo** [comment]
Same crash for me, `Accessing PalettedContainer from multiple threads`. I get routine crashes from it, every so often when I play. I can upload logs if desired.

**Comment by impo** [comment]
> Glad someone else can reproduce, I've almost given up trying. Does trashslot misbehave for you too post-crash @impo ?

Nope, trash slot behaves as normal.

**Comment by ukmcplyr** [comment]
Suspected as much, cheers for confirming!

**Comment by Raidobw2** [comment]
Thanks for the detailed report, I appreciated it!

**Comment by ukmcplyr** [comment]
No worries, glad it was useful. Unrealted but I've been chasing at least one other bug in FTB Backups 2. Is it more helpful for you guys if I report to them upstream and create a tracker bug here, or keep doing as I have been?

**Comment by perennialpersephone** [comment]
Thought I'd drop this crash log here as I am also getting an occasional "**Accessing PalettedContainer from multiple threads**"

Mine was caused by a player stacking Watches of flowing time...
[crash-2022-11-18_13.30.01-server.txt](https://github.com/FTBTeam/FTB-Modpack-Issues/files/10046250/crash-2022-11-18_13.30.01-server.txt)

**Comment by perennialpersephone** [comment]
@ukmcplyr
 ```-- Entity being ticked --
Details:
	Entity Type: create:stationary_contraption (com.simibubi.create.content.contraptions.components.structureMovement.ControlledContraptionEntity)
	Entity ID: 7
	Entity Name: Stationary Contraption
	Entity's Exact location: 58.00, 112.00, 0.00
	Entity's Block location: World: (58,112,0), Section: (at 10,0,0 in 3,7,0; chunk contains blocks 48,-128,0 to 63,127,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-128,0 to 511,127,511)
	Entity's Momentum: 0.00, 0.00, 0.00
	Entity's Passengers: []
	Entity's Vehicle: 
```

	
```
	-- Performance stats --
Details:
	Random tick rate: 3
	Level stats: ResourceKey[minecraft:dimension / minecraft:overworld]: players: 0, entities: 2,0,2,529,0,0,529 [], block_entities: 22 [<null>:22], block_ticks: 0, fluid_ticks: 0, chunk_source: Chunks[S] W: 0 E: 2,0,2,529,0,0,529,
ResourceKey[minecraft:dimension / ae2:spatial_storage]: players: 0, entities: 0,0,0,0,0,0,0 [], block_entities: 0 [], block_ticks: 0, fluid_ticks: 0, chunk_source: Chunks[S] W: 0 E: 0,0,0,0,0,0,0,
ResourceKey[minecraft:dimension / compactmachines:compact_world]: players: 0, entities: 0,0,0,0,0,0,0 [], block_entities: 0 [], block_ticks: 0, fluid_ticks: 0, chunk_source: Chunks[S] W: 0 E: 0,0,0,0,0,0,0,
ResourceKey[minecraft:dimension / minecraft:the_nether]: players: 0, entities: 0,0,0,0,0,0,0 [], block_entities: 0 [], block_ticks: 0, fluid_ticks: 0, chunk_source: Chunks[S] W: 0 E: 0,0,0,0,0,0,0,
ResourceKey[minecraft:dimension / ftbsbc:team/ephods-cheapens-rungs]: players: 0, entities: 6,0,5,625,0,0,625 [], block_entities: 345 [functionalstorage:compacting_drawer:24,create:mechanical_crafter:21,functionalstorage:oak_1:21,minecraft:campfire:20,create:simple_kinetic:18], block_ticks: 109, fluid_ticks: 0, chunk_source: Chunks[S] W: 2401 E: 6,0,5,625,0,0,625,
ResourceKey[minecraft:dimension / bloodmagic:dungeon]: players: 0, entities: 0,0,0,0,0,0,0 [], block_entities: 0 [], block_ticks: 0, fluid_ticks: 0, chunk_source: Chunks[S] W: 0 E: 0,0,0,0,0,0,0,
ResourceKey[minecraft:dimension / minecraft:the_end]: players: 0, entities: 0,0,0,0,0,0,0 [], block_entities: 0 [], block_ticks: 0, fluid_ticks: 0, chunk_source: Chunks[S] W: 0 E: 0,0,0,0,0,0,0
```
	From your crash logs, it appears these blocks are being tick accelerated? Do you have watches of flowing time nearby? #2088 Also seems to be having a crash due to tick acceleration, My own crash log posted above is also due to tick acceleration... (Watches of flowing time in my case and 2088) so I wonder if Project E and FTBBackups aren't playing nice with one another?

**Comment by ukmcplyr** [comment]
Nope, just the bost basic Create tree farm that's incredibly slow due to the minum amount of wind sails underneath. Can screenshot when I'm in front of the game if needed :)

**Comment by ukmcplyr** [comment]
I'm pretty certain this bug has nothing to do with tick accelerating though doing so could well increase the chances of it happening.

I did a fairly deep dive and from what I found the bug basically comes down to:

1. FTB Backups 2 or code it calls into uses PalettedContainer
2. PalettedContainer is not thread safe
3. Access to PalettedContainer is not wrapped around a mutex or other lock

From memory, at least one other mod had to workaround this, iirc it was one of YUNG's mods (maybe Better Dungeons).
